### PR TITLE
drop use of BIO_f_base64 and EVP_PKEY_new_mac_key

### PR DESCRIPTION
**Pull request by randall**
BIO_f_base64 is not available in BoringSSL plus we have our own Base64
methods. With this change, the estimation values change, but shouldn't
effect actual runtime.

And BoringSSL suggests using the HMAC* functions in place of EVP_PKEY_new_mac_key

### CODE REVIEW

**Comment by randall** [comment]
[approve ci autest]

File: plugins/experimental/access_control/unit_tests/test_utils.cc [hunk 1]
```diff
@@ -26,34 +26,33 @@
...
 
 TEST_CASE("Base64: estimate buffer size needed to encode a message", "[Base64][access_control][utility]")
 {
   size_t encodedLen;
 
-  /* Test with a zero decoded message lenght */
+  /* Test with a zero decoded message length */
   encodedLen = cryptoBase64EncodedSize(0);
-  CHECK(0 == encodedLen);
+  CHECK(4 == encodedLen);
```
Comment (maskit) [inline plugins/experimental/access_control/unit_tests/test_utils.cc:39]: This is a bit sad. Seems like the comment in ink_base64.h is not true. We may want to improve the macro first. Of course we can last this first and change back these values to the original values after the improvement though.

> Little helper functions to calculate minimum required output buffer for encoding/decoding.
> #define ATS_BASE64_ENCODE_DSTLEN(_length) ((_length * 8) / 6 + 4)

File: plugins/experimental/access_control/unit_tests/test_utils.cc [hunk 2]
```diff
@@ -180,8 +180,6 @@ TEST_CASE("Base64: quick encode / decode with '+', '/' and various paddings", "[
     CHECK(0 == strncmp(encodedMessage, encoded[i], encodedMessageLen));
 
     /* Decode */
-    size_t decodedMessageEstimatedLen = cryptoBase64DecodeSize(encodedMessage, encodedMessageLen);
-    CHECK(strlen(decoded[i]) == decodedMessageEstimatedLen);
```
Comment (maskit) [inline plugins/experimental/access_control/unit_tests/test_utils.cc:184]: Why was this check deleted?

**Comment by randall** [comment]
[approve ci autest]

File: plugins/experimental/access_control/unit_tests/test_utils.cc (hunk 2 above)
Comment (randall) [reply plugins/experimental/access_control/unit_tests/test_utils.cc:184]: It goes back how `cryptoBase64DecodeSize` determines its size for input. The test here assumes the estimate is not an estimate and rather an accurate value.

here's the run with the check in place (with some extra debugging):
```
experimental/access_control/unit_tests/test_utils.cc:186: FAILED:
  CHECK( strlen(decoded[i]) == decodedMessageEstimatedLen )
with expansion:
  11 == 12
with messages:
  i := 1
  decoded[i] := "ts>ts?ts!!!"

experimental/access_control/unit_tests/test_utils.cc:186: FAILED:
  CHECK( strlen(decoded[i]) == decodedMessageEstimatedLen )
with expansion:
  10 == 12
with messages:
  i := 2
  decoded[i] := "ts>ts?ts!!"
```

**Comment by randall** [comment]
[approve ci docs]

**Comment by maskit** [comment]
> And BoringSSL suggests using the HMAC* functions in place of EVP_PKEY_new_mac_key

HMAC* functions are going to be deprecated since OpenSSL 3.0.

I'd like to land this before making changes for OpenSSL 3.0. How about commenting out the test case instead of removing so that we can restore it easily?

**Comment by bryancall** [comment]
[approve ci]

**Comment by bryancall** [comment]
[approve ci docs]

**Comment by randall** [comment]
> > And BoringSSL suggests using the HMAC* functions in place of EVP_PKEY_new_mac_key
> 
> HMAC* functions are going to be deprecated since OpenSSL 3.0.
> 
> I'd like to land this before making changes for OpenSSL 3.0. How about commenting out the test case instead of removing so that we can restore it easily?

Commented out, instead of deleting it.

**Review by maskit — APPROVED** [review]
Thank you for updating.

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**